runner/internal/shim/docker.go

```diff
@@ -82,10 +85,43 @@ func (d *DockerRunner) Run(ctx context.Context, cfg DockerImageConfig) error {
 
 	log.Printf("Container finished successfully, id=%s\n", containerID)
 
+	d.currentContainer = ""
 	d.state = Pending
 	return nil
 }
 
+func (d *DockerRunner) SubmitStop(force bool) {
+	if d.currentContainer == "" {
+		return
+	}
+
+	ctx := context.Background()
+
+	stopOptions := container.StopOptions{}
+	if force {
+		timeout := int(0)
+		stopOptions.Timeout = &timeout
+	}
+
+	err := d.client.ContainerStop(ctx, d.currentContainer, stopOptions)
+	if err != nil {
+		log.Printf("Failed to stop container: %s", err)
+	}
+
+	if !d.dockerParams.DockerKeepContainer() {
```

Comment:
Isn't it a duplication of `defer` inside of the `Run` method?

Reply:
Yes, it is! Thanks! Fixed.